File: uva11827.c
int max(int a, int b){
	if(a>b)return a;
	else return b;
	}

int	GCD( int a, int b ) {
	int		r;
	while ( b ) {
		r = a % b;
		a = b;
		b = r;
	}

	return a;
}


int main(){

int t,i=0,j=0,n[101],k=0,a=0,b=0;
char c[200];

scanf("%d",&t);
getchar();

while(t--){
    gets(c);

    i = j = k = a = b = n[0] = 0 ;

    while(i<200){
        if(c[i]=='\0')
            break;

        if(c[i] == ' '){
            j = 0;
            
            if (a){
				k++;
				n[k] = 0 ;
				}
            a = 0;
        }

        else{
            n[k]= n[k]*10 + c[i]-'0';
            a=1;
		}
        i++;
    }
    
    j = 1;
    for (a = 0; a <= k ; a++){
        for(b = a+1; b <= k ;b++){
                j = max(GCD(n[a],n[b]),j);
        }
    }
    printf("%d\n",j);

}

return 0;
}

File: uva11827.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;


public class uva11827 {
	
	static int GCD(int a, int b){
		int r;
		while(b>0){
			r=a%b;
			a=b;
			b=r;
		}
		return a;
	} 
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer tk;
		
		int n[] = new int[101];
		int t = Integer.parseInt(br.readLine());
		while(t-->0){
			tk = new StringTokenizer(br.readLine());
			int i = tk.countTokens();
			int j =0 ;
			for (j = 0  ; j < i; j++) 
				n[j] = Integer.parseInt(tk.nextToken());
			
			int ans = 1;
			for (int a = 0; a < j ; a++){
		        for(int b = a+1; b < j ;b++){
		                ans = Math.max(GCD(n[a],n[b]),ans);
		        }
		    }
			System.out.println(ans);
		}
		
	}

}
